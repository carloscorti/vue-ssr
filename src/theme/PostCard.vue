<template>
  <div class="card">
    <div class="card-content">
      <slot name="title"></slot>
      <slot name="content"></slot>
    </div>
    <footer class="card-footer">
      <a class="card-footer-item" :href="post.link" target="_blank"
        >Read More</a
      >
    </footer>
  </div>
</template>

<script>
export default {
  name: 'PostCard',
  props: {
    post: {
      type: Object,
      validator: prop => {
        return Object.entries(prop).every(value => {
          if (value[0] === 'id') {
            return typeof value[1] === 'number' && value[1] >= 0;
          }
          return typeof value[1] === 'string';
        });
      }
    }
  }
};
</script>

<style lang="scss" scoped>
.card {
  padding-bottom: 40px;
  height: 100%;
}
footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  left: 0;
}
</style>
