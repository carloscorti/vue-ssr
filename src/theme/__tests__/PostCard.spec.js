import { mount, shallowMount } from '@vue/test-utils';
import PostCard from '../PostCard.vue';

const testPost = {
  id: 0,
  title: { rendered: 'this is a test post' },
  content: { rendered: 'test post content' },
  rest_api_enabler: { Link: '/#' }
};

describe('PostCard', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(PostCard, {
      propsData: {
        post: testPost
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('loads post prop correctly', () => {
    const wrapper = mount(PostCard, {
      propsData: {
        post: testPost
      }
    });
    expect(wrapper.props().post).toBe(testPost);
  });
});
