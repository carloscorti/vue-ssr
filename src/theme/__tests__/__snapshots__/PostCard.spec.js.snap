// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard renders correctly 1`] = `
<div class="card">
  <div class="card-content"> </div>
  <footer class="card-footer"><a href="/#" target="_blank" class="card-footer-item">Read More</a></footer>
</div>
`;
